pkg/storage/tsdb/bucketindex/updater.go

```diff
@@ -109,12 +109,22 @@ func (w *Updater) updateBlocks(ctx context.Context, old []*Block) (blocks []*Blo
 		}
 
 		if errors.Is(err, ErrBlockMetaNotFound) {
-			partials[id] = err
+			partials = append(partials, PartialBlock{
+				ID:      id,
+				MinTime: b.MinTime,
+				MaxTime: b.MaxTime,
+				Reason:  err,
```

Comment:
`b.MinTime` and `b.MaxTime` will be 0. They are stored in `meta.json` file, but we failed to read that file.

Given that, let's revert changes to `bucketindex` package -- getting `map[ulid]error` is perfectly fine for our purposes.